cardano-api/src/Cardano/Api/Orphans.hs

```diff
@@ -0,0 +1,245 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+
+module Cardano.Api.Orphans () where
+
+import           Prelude
+
+import           Control.Iterate.SetAlgebra (BiMap (..), Bimap)
+import           Data.Aeson (ToJSON (..), object, (.=))
+import qualified Data.Aeson as Aeson
+import           Data.Aeson.Types (ToJSONKey (..), toJSONKeyText)
+import qualified Data.ByteString.Base16 as B16
+import qualified Data.Map.Strict as Map
+import           Data.Text (Text)
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+
+import qualified Cardano.Crypto.Hash.Class as Crypto
+import qualified Cardano.Ledger.Core as Core
+import qualified Cardano.Ledger.Crypto as Crypto
+import qualified Cardano.Ledger.Mary.Value as Mary
+import           Cardano.Slotting.Slot (SlotNo (..))
+import qualified Ouroboros.Consensus.Shelley.Eras as Consensus
+import qualified Shelley.Spec.Ledger.API as Shelley
+import           Shelley.Spec.Ledger.BaseTypes (StrictMaybe)
+import qualified Shelley.Spec.Ledger.Coin as Shelley
+import qualified Shelley.Spec.Ledger.Delegation.Certificates as Shelley
+import qualified Shelley.Spec.Ledger.EpochBoundary as ShelleyEpoch
+import qualified Shelley.Spec.Ledger.LedgerState as ShelleyLedger
+import qualified Shelley.Spec.Ledger.Rewards as Shelley
+
+-- Orphan instances involved in the JSON output of the API queries.
+-- We will remove/replace these as we provide more API wrapper types
```

Comment:
Mmm. This does make me a little nervous. Where do we still need these? Is this for the ToJSON instance for the ledger state? Otherwise I think we should be able to do everything with JSON instances only for the API surface types.

Reply:
Yes these are for the `LedgerState` and `ProtocolState` which are just type wrappers around the underlying ledger-specs types.